hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/AbstractBackendStoreProvider.java

```diff
@@ -152,12 +152,15 @@ public void initSystemInfo(HugeGraph graph) {
 
     @Override
     public void writeSnapshot() {
-        // TODO: to be implemented
+        String snapshotPrefix = "snapshot";
+        for (BackendStore store : this.stores.values()) {
+            store.writeSnapshot(snapshotPrefix);
+        }
```

Comment:
prefer call store.writeSnapshot() through tx, and remove writeSnapshot() from AbstractBackendStoreProvider class

Reply:
Only memory backend will share store provider